OmniAuth Login to GitHub

maresTECH opened the pull request:
at localhost:3000, you should be prompted with a 'login to Github' link.  OmniAuth should take you to the site to verify yourself, and them redirect you to the employee db.  While there you can use the application or click the 'logout' link to be redirected to the initial login screen.  You will not be able to access the db (localhost:3000/employees) without being logged in so the user MUST have a GitHub account to view.

Nothing flashy yet about this yet, just mildly functional!

maresTECH committed:
added omniauth gem bundle

maresTECH committed:
Add Session controller

  $rails g controller Session create destroy

This will help create and login and logout.

maresTECH committed:
Add link to authenticate

  $= link_to 'Login with GitHub', developer_auth_url

added this link to login using OAuth.

maresTECH committed:
Modify session controller test

  changed:
  $get session_change_url

  to:
  $get developer_auth_url

maresTECH committed:
Create User model

  $rails generate model User uid:string provider:string name:string

maresTECH committed:
Modify session controller

after user logs into app, user is rerouted to employee db.

maresTECH committed:
Add destroy

  $reset_session
  $
  $redirect_to :root

  This essentially is the logout function to reset a users session
  and to redirect user back to the login.  I have added the logout
  link to views/employees/index.

maresTECH committed:
Modify application and employees controllers

defined three methods to make sure a user must be logged in before
they can access the employee db.

maresTECH committed:
Modify login with github omniauth

adjusted omniauth login link to redirect at github.  Users will now
use their github accounts in order to login and view the employee db.

maresTECH committed:
Modify session controller test

Preparing dev branch for merge with master branch.

tarunngidwani reviewed (changes requested):
1) Remove omniauth gem from Gemfile

2) Add a strict versioning for the omniauth-github gem

3) Move omniauth-gem to the bottom of the Gemfile. Put it is alongside my custom gem additions.

4) Changing the root route should be a separate commit

tarunngidwani reviewed (changes requested):
1) Need to run rails generate controller with more restrictive options, as we do not need coffee, scss, helper, files, or a create and destroy view

Look at rails g controller --help, to view the options

tarunngidwani reviewed (changes requested):
1) The login view needs to be in app/views/sessions not part of employees view

2) Need to move this to an initializer file in config/initialziers directory. Also need a condition to set the appropriate provider based on environment production or dev

3) Route file overall requires clean up, no need for a match via post there. Also need to fix he old usage of hashes

tarunngidwani reviewed (changes requested):
No skip before action needed. If we set stuff up correctly

No need to worry about tests. I will be adding the tests

tarunngidwani reviewed (changes requested):
Talked to the professor. There is no requirement of having a User model

So this can be removed.

tarunngidwani reviewed (changes requested):
Not required to save user information. We can remove that.

Any information we do need to save we can create a current user hash and store that in session so we can access is everywhere else

tarunngidwani reviewed (changes requested):
We will redirect to the login page. Which should be named login

The logout button should be placed in applicaiton.html.haml, so it is visible on every page.

Use new hash syntax for naming destroy route

maresTECH committed:
Clean up Gemfile

  - Remove omniauth gem

  - Specify version for omniauth github gem

maresTECH committed:
Clean up gemfile

  - Remove omniauth gem

tarunngidwani reviewed (approved):
Good clean up

tarunngidwani reviewed (approved):
Looks good. Cleaning up additional remaining omniauth entry in Gemfile

maresTECH committed:
Delete unnecessary asset and helper files

  - Remove files generated by the rail generator that are not needed

tarunngidwani reviewed (approved):
Removing files that are not need but were created by the rails g controller generator

maresTECH committed:
Move login.html.haml

  - Move from views/employees to views/sessions

maresTECH committed:
Move omniauth initialization

  - Move from application.rb to omniauth initialization file
    under config/initializers

tarunngidwani reviewed (approved):
Move to initializer file. Nice

maresTECH committed:
Modify routes

  - Update root route to sessions#login action

  - Update routes to use new hash syntax

  - Create get route to auth callback

maresTECH committed:
Update session to sessions

maresTECH committed:
Clean up session controller and test

  - Remove unnecessary methods and actions

  - Remove non functioning test

maresTECH committed:
Changed colen to semi-colen

maresTECH committed:
Remove User model

  - Remove all files generated by rails g model generator
    related to the user model

tarunngidwani reviewed (approved):
Cleaning up user model artifacts. Not part of beta app requirements

maresTECH committed:
Create sessions#create action

maresTECH committed:
Fix errors breaking github login

  - Remove before action left over from user model

  - Rename file and class for controller from session to sessions

  - Remove unnecessary omniauth config from application config file

tarunngidwani reviewed (approved):
Complete feature: Giving users the ability to login to our application using their Github credentials

maresTECH committed:
Remove non functioning logout feature

  - Create empty session#destroy action

  - Remove logout link from employees index page

  - Remove get logout route from routes file

tarunngidwani reviewed (approved):
Good clean up for non functioning logout feature
